### ISSUE TITLE
SDK Build Error

### ISSUE THREAD

**Issue by Yuhanlolo** [alexa/avs-device-sdk#499]
**IMPORTANT**: Before you create an issue, please take a look at our [Issue Reporting Guide](https://github.com/alexa/avs-device-sdk/wiki/Issue-Reporting-Guide).

## Briefly summarize your issue:

I'm following this instruction: https://github.com/alexa/avs-device-sdk/wiki/macOS-Quick-Start-Guide, and got to step 2.2, which runs the command "make SampleApp -j2". I also tried "-j3", "j4", and "make", but kept getting the same error after 93% of building completed.

## What is the expected behavior?
I guess it suppose to complete 100%.

## What behavior are you observing?
[ 93%] Built target SpeechSynthesizer
[ 93%] Built target AVSSystem
[ 93%] Built target DefaultClient
[ 93%] Built target AuthDelegate
[ 93%] Built target PlaylistParser
[ 93%] Built target MediaPlayer
make[3]: *** No rule to make target `/Users/Yuhan/sdk-folder/third-party/portaudio/lib/.libs/libportaudio.a', needed by `SampleApp/src/SampleApp'.  Stop.
make[2]: *** [SampleApp/src/CMakeFiles/SampleApp.dir/all] Error 2
make[1]: *** [SampleApp/src/CMakeFiles/SampleApp.dir/rule] Error 2

## Provide the steps to reproduce the issue, if applicable:
I just followed the instruction. Every step before went well.

### What version of the AVS Device SDK are you using?
AVS SDK V1.4

### Tell us what hardware you're using:
Desktop / Laptop

### Tell us about your OS (Type & version):
MacOS 

I saw other people post similar issues but they are using different OS, and I didn't see the solution either. I have been trying to reinstalled my gcc but it didn't help.

**Comment by mradulan** [comment]
do you have libportaudio.a in the path /Users/Yuhan/sdk-folder/third-party/portaudio/lib/.libs/?
Can you do a `ls` in the path  /Users/Yuhan/sdk-folder/third-party/portaudio/lib/.libs/

**Comment by Yuhanlolo** [comment]
Hi, thanks for the response! Yes you are right I don't have libportaudio.a in the path /Users/Yuhan/sdk-folder/third-party/portaudio/lib/.libs/. I only have a portaudio folder under third-party but there's no lib folder inside it. So I realized that the step before "make" didn't run successfully, either. 

I searched some relevant posts and some people say disabling parallel build might help, so I did disabled it and re ran the cmake, but it still didn't work.

**Comment by mradulan** [comment]
Inside portaudio directory, you have to run ./configure && make. This will create libportaudio.a and then you can build the SDK again with cmake option.

**Comment by mradulan** [comment]
http://portaudio.com/docs/v19-doxydocs/compile_mac_coreaudio.html

**Comment by Yuhanlolo** [comment]
Thank you! I think it works. The output shows it's built successfully, though not 100%, and there's no error log:
[ 93%] Linking CXX executable SampleApp
[ 93%] Built target SampleApp

Although I got some error when running ./configure && make, I was able to figure it out. I'll put my steps here in case others got the same issues:

error: "tool 'xcodebuild' requires Xcode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance."
solution: run command: "sudo xcode-select --switch /Library/Developer/CommandLineTools"

error: "Could not find 10.5 to 10.12 SDK."
solution: download MacOS SDK 10.12 [here](https://github.com/phracker/MacOSX-SDKs/releases), and put it in "/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs"
